src/index.js.flow

```diff
@@ -120,24 +120,24 @@ declare export function createRouterContext(
  * Utility method to created async versions of getData functions
  *
  */
-type GetDataLoader = () => Promise<{
-  default: $PropertyType<RouteResource, 'getData'>,
+type GetDataLoader<T> = () => Promise<{
+  default: $PropertyType<RouteResource<T>, 'getData'>,
   ...
 }>;
 
 /**
  * Utility method to created type safe resources with defaults.
  *
  */
-type CreateResourceArg =
-  | {| ...RouteResource, maxAge?: number |}
+type CreateResourceArg<T> =
+  | {| ...RouteResource<T>, maxAge?: number |}
   | {|
       ...$Diff<
-        RouteResource,
-        {| getData: $ElementType<RouteResource, 'getData'> |}
+        RouteResource<T>,
+        {| getData: $PropertyType<RouteResource<T>, 'getData'> |}
       >,
       maxAge?: number,
-      getDataLoader: GetDataLoader,
+      getDataLoader: GetDataLoader<T>,
     |};
 
-declare export function createResource(args: CreateResourceArg): RouteResource;
+declare export function createResource<T>(args: CreateResourceArg<T>): RouteResource<T>;
```

Comment:
Note that this makes `createResource` _require_ a type argument. It won't default to `any` and Flow doesn't seem to be able to infer the type based on `getData`/`getDataLoader`. However, the conversion is fairly straight forward; it just takes the type of the result that `getData`'s returned Promise returns.